### ISSUE TITLE
Race condition bug triggered by step failure

### ISSUE THREAD

**Issue by vhadianto** [turbot/flowpipe#630]
**Describe the bug**
This is an intermittent bug. 

Sometime this test case:  https://github.com/turbot/flowpipe/blob/v0.2.2/internal/es/estest/mod_test.go#L1414

fails, the log is indicating that the step hasn't been retried before a pipeline fail has been raised. What **should** happen here is the step retry:

```
    step "http" "bad_http" {
        url = "http://api.google.com/astros.jsons"

        retry {
            if = result.status_code == 404
            max_attempts = 3
        }
    }
```

should be fully completed before the pipeline fail is raised.

There is a race condition somehwere.

**Flowpipe version (`flowpipe -v`)**
Example: v0.2.0

**Comment by vhadianto** [comment]
Events dump from one of the failed run:

[551-race-condition.json](https://github.com/turbot/flowpipe/files/14158689/551-race-condition.json)

**Comment by vhadianto** [comment]
Fixed with https://github.com/turbot/flowpipe/commit/907ef7d4e96cd0ee95c6d1b29122829d03364aba

Released with v0.2.3
